botnim/query.py

```diff
@@ -221,9 +217,20 @@ def format_search_results(results: List[SearchResult], format: str, explain: boo
             if explain and hasattr(result, '_explanation'):
                 result_dict['_explanation'] = result._explanation
             formatted_results.append(result_dict)
+        elif format == 'yaml':
+            # For YAML, split header/text for each result
+            parts = result.full_content.split('\n\n', 1)
+            result_dict = dict(
+                header=parts[0].strip(),
+                text=parts[1].strip() if len(parts) > 1 else '',
+            )
+            formatted_results.append(result_dict)
     
     if join:
         formatted_results = '\n'.join(formatted_results)
+    if format == 'yaml':
+        import yaml
```

Comment:
no need for this import, already imported above. generally it's bad practice to import inside a method.